lib/class-wp-theme-json-gutenberg.php

```diff
@@ -606,7 +606,8 @@ public function __construct( $theme_json = array(), $origin = 'theme' ) {
 		$registry            = WP_Block_Type_Registry::get_instance();
 		$valid_block_names   = array_keys( $registry->get_all_registered() );
 		$valid_element_names = array_keys( static::ELEMENTS );
-		$theme_json          = static::sanitize( $this->theme_json, $valid_block_names, $valid_element_names );
+		$valid_variations    = array();
```

Comment:
Shouldn't this array be populated at some point with the valid variations for each block? 

If there are no unregistered variations, on trunk adding styles to an existing variation both in theme.json and through the global styles UI works as expected, and styles display both in the post editor and on the front end. In this branch, neither are working.

Reply:
:see_no_evil: 

I've fixed, simplified and added a test for this. It should be ready now.